### PR TITLE
displayComponentNames

### LINKED ISSUE #158: Add component names on icon layer
According to Modelica programming guidelines, the names of component instances should be added to the icon layer of each component.

This helps the modeler to quickly indentify different components in complex fluid networks.

In addition, if there are relevant parameters or variables of certain components, they should also be displayed in the icon layer. This is also done in some components of the MSL (written in black):

### PR DESCRIPTION
**Pull request by RaphaelGebhart**
Added displayComponentNames to dropOfCommons, added Utilities.DisplayComponentNameIndividually which all classes shall extend, added 'name' text to 'most' components (Bidirectional missing, and some components where we would have to move the connectors) closes #158

### CODE REVIEW

**Comment by RaphaelGebhart** [comment]
the `extends ThermofluidStream.Utilities.DisplayComponentNameIndividually;` doesnt work unfortunatly... so i either have do it without extends, i.e.  `parameter Boolean displayComponentName = dropOfCommons.displayComponentNames` for every component... or i might move dropOfCommons to `ThermofluidStream.Utilities.DisplayComponentNameIndividually` ....

**Comment by nieweber** [comment]
I'd personally prefer to add the parameter `displayComponentName ` to each component or base class (`ThermofluidStream.Interfaces.SISOFlow`,...) if possible...

Another thing that we have to take care of is that there are still some components that don't have an outer connection to `dropOfCommons` (for example `ThermofluidStream.Boundaries.TerminalSource`) and hence they don't get the parameter.

**Comment by RaphaelGebhart** [comment]
OK then i add it to every component, but please check if `parameter Boolean displayComponentName = dropOfCommons.displayComponentNames "=true, if u wish to display the component name" annotation(Dialog(group="Layout"),Evaluate=true, HideResult=true); ` because i have to copy paste that to 100 components then :D
`

**Comment by RaphaelGebhart** [comment]
> Another thing that we have to take care of is that there are still some components that don't have an outer connection to `dropOfCommons` (for example `ThermofluidStream.Boundaries.TerminalSource`) and hence they don't get the parameter.

yes i already did that

**Comment by RaphaelGebhart** [comment]
> OK then i add it to every component, but please check if `parameter Boolean displayComponentName = dropOfCommons.displayComponentNames "=true, if u wish to display the component name" annotation(Dialog(group="Layout"),Evaluate=true, HideResult=true); ` is fine, because i have to copy paste that to 100 components then :D `

since there are quite some components that dont use `ThermofluidStream.Interfaces.SISOFlow` ....

**Comment by RaphaelGebhart** [comment]
Regression tests passed. Still to be done (besides reviewing the changes):

- At the moment `displayInstanceName(s) = false` and `displayParameters = false` for all Tests/Examples. If you like you could change if beneficial.

- With the last commit we should (re)move `ThermofluidStream.TestIcons` and `ThermofluidStream.TestIconParameters`

**Review by nieweber — APPROVED** [review]
Fine for me now.

File: ThermofluidStream/Boundaries/DynamicPressureOutflow.mo [hunk 1]
```diff
@@ -1,45 +1,87 @@
...
           extent={{-20,-20},{20,20}},
-        rotation=270,
-        origin={-60,100}), iconTransformation(extent={{-20,-20},{20,20}},
-        rotation=270,
-        origin={-60,100})));
+        rotation=0,
+        origin={-120,-60}),
+                         iconTransformation(extent={{-20,-20},{20,20}},
+        rotation=0,
+        origin={-120,-60})));
```
Comment (tobolar) [inline ThermofluidStream/Boundaries/DynamicPressureOutflow.mo:66]:
```suggestion
        origin={-120,-60})));
```

File: ThermofluidStream/Boundaries/Internal/PartialVolume.mo [hunk 2]
```diff
@@ -54,40 +60,38 @@ inlets and outlets the volume is connected to.
...
-  SI.Mass MXi[Medium.nXi](each stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = M*medium.Xi;
-  SI.Energy U_med(stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = M*medium.u;
+  SI.Mass M(stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = V*medium.d "Mass";
+  SI.Mass MXi[Medium.nXi](each stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = M*medium.Xi "Masses of species";
+  SI.Energy U_med(stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = M*medium.u "Internal energy";
 
-  SI.HeatFlowRate Q_flow;
-  SI.Power W_v;
+  SI.HeatFlowRate Q_flow "Heat flow rate";
+  SI.Power W_v "Volumenänderungsarbeitsstrom (work due to change of volume)";
```
Comment (tobolar) [inline ThermofluidStream/Boundaries/Internal/PartialVolume.mo:71]: Dont use GE in parameter description. Suggestion:
```suggestion
  SI.Power W_v "Work due to change of volume";  // Volumenaenderungsarbeitsstrom
```

File: ThermofluidStream/Boundaries/Internal/PartialVolumeM.mo [hunk 3]
```diff
@@ -48,38 +55,36 @@ inlets and outlets the volume is connected to.
...
-  SI.Mass MXi[Medium.nXi](each stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = M*medium.Xi;
-  SI.Energy U_med(stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = M*medium.u;
+  SI.Mass M(stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = V*medium.d "Mass";
+  SI.Mass MXi[Medium.nXi](each stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = M*medium.Xi "Masses of species";
+  SI.Energy U_med(stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = M*medium.u "Internal energy";
 
-  SI.HeatFlowRate Q_flow;
-  SI.Power W_v;
+  SI.HeatFlowRate Q_flow "Heat flow rate";
+  SI.Power W_v "Volumenänderungsarbeitsstrom (work due to change of volume)";
```
Comment (tobolar) [inline ThermofluidStream/Boundaries/Internal/PartialVolumeM.mo:66]: Don't use GE in parameter description. Suggestion:
```suggestion
  SI.Power W_v "Work due to change of volume";  // Volumenaenderungsarbeitsstrom
```

File: ThermofluidStream/Boundaries/Internal/PartialVolumeN.mo [hunk 4]
```diff
@@ -49,38 +54,36 @@ inlets and outlets the volume is connected to.
...
-  SI.Mass MXi[Medium.nXi](each stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = M*medium.Xi;
-  SI.Energy U_med(stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = M*medium.u;
+  SI.Mass M(stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = V*medium.d "Mass";
+  SI.Mass MXi[Medium.nXi](each stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = M*medium.Xi "Masses of species";
+  SI.Energy U_med(stateSelect=if usePreferredMediumStates then StateSelect.default else StateSelect.always) = M*medium.u "Internal energy";
 
-  SI.HeatFlowRate Q_flow;
-  SI.Power W_v;
+  SI.HeatFlowRate Q_flow "Heat flow rate";
+  SI.Power W_v "Volumenänderungsarbeitsstrom (work due to change of volume)";
```
Comment (tobolar) [inline ThermofluidStream/Boundaries/Internal/PartialVolumeN.mo:65]: Don't use GE in parameter description. Suggestion:
```suggestion
  SI.Power W_v "Work due to change of volume";  // Volumenaenderungsarbeitsstrom
```

File: ThermofluidStream/Boundaries/PhaseSeparator2.mo [hunk 5]
```diff
@@ -90,8 +90,12 @@ equation
...
   state_out[2] = Medium.setState_phX(medium.p, h_pipe2, medium.Xi);
 
-  annotation (Icon(coordinateSystem(preserveAspectRatio=false)),
-                            Diagram(coordinateSystem(preserveAspectRatio=false)),
+  annotation (Icon(coordinateSystem(preserveAspectRatio=true),graphics={
+       Text(visible=displayInstanceName,
+          extent={{-150,140},{150,100}},
+          textString="%name",
+          textColor=dropOfCommons.instanceNameColor)}),
+                            Diagram(coordinateSystem(preserveAspectRatio=true)),
```
Comment (tobolar) [inline ThermofluidStream/Boundaries/PhaseSeparator2.mo:98]:
```suggestion
    Diagram(coordinateSystem(preserveAspectRatio=true)),
```

File: ThermofluidStream/Boundaries/Tests/Reservoir.mo [hunk 6]
```diff
@@ -116,7 +116,8 @@ equation
       points={{44,-56},{36,-56}},
       color={28,108,200},
       thickness=0.5));
-  connect(ramp.y, reservoir.pEnv_input) annotation (Line(points={{-59,-18},{-4,-18},{-4,-46}}, color={0,0,127}));
+  connect(ramp.y, reservoir.pEnv_input) annotation (Line(points={{-59,-18},{-4,-18},{-4,-44.1}},
+                                                                                               color={0,0,127}));
```
Comment (tobolar) [inline ThermofluidStream/Boundaries/Tests/Reservoir.mo:120]:
```suggestion
  connect(ramp.y, reservoir.pEnv_input) annotation (Line(points={{-59,-18},{-4,-18},{-4,-44.1}}, color={0,0,127}));
```

File: ThermofluidStream/Examples/SimpleAirCycle.mo [hunk 7]
```diff
@@ -166,39 +166,39 @@ initial equation
...
   connect(splitterT1_1.outletA, heatExchange_CounterFlowNTU1.inletB)
     annotation (Line(
-      points={{-10,-40},{-54,-40},{-54,-31.2}},
+      points={{-10,-40},{-54,-40},{-54,-30}},
       color={28,108,200},
       thickness=0.5));
   connect(splitterT1_1.outletB, heatExchange_CounterFlowNTU2.inletB)
     annotation (Line(
-      points={{10,-40},{54,-40},{54,-1.2}},
+      points={{10,-40},{54,-40},{54,5.32907e-15}},
```
Comment (tobolar) [inline ThermofluidStream/Examples/SimpleAirCycle.mo:201]:
```suggestion
      points={{10,-40},{54,-40},{54,0}},
```

File: ThermofluidStream/Examples/SimpleAirCycle.mo [hunk 8]
```diff
@@ -246,7 +246,7 @@ equation
         points={{56,-60},{40,-60},{40,50}},
         color={0,0,0}));
   connect(fan1.inlet, heatExchange_CounterFlowNTU2.outletA) annotation (Line(
-      points={{66,-50},{66,-1}},
+      points={{66,-50},{66,1.77636e-15}},
```
Comment (tobolar) [inline ThermofluidStream/Examples/SimpleAirCycle.mo:249]:
```suggestion
      points={{66,-50},{66,0}},
```

File: ThermofluidStream/Examples/SimpleEngine.mo [hunk 9]
```diff
@@ -203,7 +203,7 @@ equation
       points={{-10,44},{-10,34},{-2,34}},
       color={28,108,200},
       thickness=0.5));
-  connect(crankDrive.h_out, booleanToReal.u) annotation (Line(points={{-85.5,-32},{-85.5,-16},{-79.6,-16}}, color={255,0,255}));
+  connect(crankDrive.h_out, booleanToReal.u) annotation (Line(points={{-90,-40},{-90,-16},{-79.6,-16}},     color={255,0,255}));
```
Comment (tobolar) [inline ThermofluidStream/Examples/SimpleEngine.mo:206]:
```suggestion
  connect(crankDrive.h_out, booleanToReal.u) annotation (Line(points={{-90,-40},{-90,-16},{-79.6,-16}}, color={255,0,255}));
```

File: ThermofluidStream/Examples/SimpleEngine.mo [hunk 10]
```diff
@@ -223,7 +223,8 @@ equation
       points={{-86,16},{-70,16},{-70,54},{-60,54}},
       color={28,108,200},
       thickness=0.5));
-  connect(singleSensorSelect.value_out, PI1.u_m) annotation (Line(points={{-106,16},{-140,16},{-140,22}}, color={0,0,127}));
+  connect(singleSensorSelect.value_out, PI1.u_m) annotation (Line(points={{-104.2,16},{-140,16},{-140,22}},
+                                                                                                          color={0,0,127}));
```
Comment (tobolar) [inline ThermofluidStream/Examples/SimpleEngine.mo:227]:
```suggestion
  connect(singleSensorSelect.value_out, PI1.u_m) annotation (Line(points={{-104.2,16},{-140,16},{-140,22}}, color={0,0,127}));
```

File: ThermofluidStream/Examples/Utilities/Piston.mo [hunk 11]
```diff
@@ -49,17 +49,17 @@ model Piston "Piston model with steam on both sides"
...
         rotation=90), iconTransformation(
         extent={{-20,-20},{20,20}},
-        origin={80,20},
+        origin={100,42},
         rotation=90)));
   Modelica.Mechanics.Translational.Interfaces.Flange_a flange
-    annotation(Placement(transformation(extent={{-10,-10},{10,10}}, origin={100, -10}, rotation=0),
-      iconTransformation(extent={{-10,-10},{10,10}}, origin={100, -10}, rotation=0)));
+    annotation(Placement(transformation(extent={{-10,-10},{10,10}}, origin={100,0},    rotation=0),
+      iconTransformation(extent={{-10,-10},{10,10}}, origin={100,0},    rotation=0)));
```
Comment (tobolar) [inline ThermofluidStream/Examples/Utilities/Piston.mo:59]:
```suggestion
    annotation(Placement(transformation(extent={{-10,-10},{10,10}}, origin={100,0}, rotation=0),
      iconTransformation(extent={{-10,-10},{10,10}}, origin={100,0}, rotation=0)));
```

File: ThermofluidStream/Examples/VenturiPump.mo [hunk 12]
```diff
@@ -127,7 +127,7 @@ equation
       color={28,108,200},
       thickness=0.5));
   connect(singleFlowSensor8.inlet, source2_2.outlet) annotation (Line(
-      points={{0,60},{0,70}},
+      points={{1.77636e-15,60},{1.77636e-15,66},{0,66},{0,70}},
```
Comment (tobolar) [inline ThermofluidStream/Examples/VenturiPump.mo:130]:
```suggestion
      points={{0,60},{0,66},{0,70}},
```

File: ThermofluidStream/Examples/VenturiPump.mo [hunk 13]
```diff
@@ -145,7 +145,7 @@ equation
       thickness=0.5));
   connect(singleFlowSensor8.outlet, dynamicPressureInflow2.inlet)
     annotation (Line(
-      points={{0,40},{0,30}},
+      points={{-1.77636e-15,40},{-1.77636e-15,36},{0,36},{0,30}},
```
Comment (tobolar) [inline ThermofluidStream/Examples/VenturiPump.mo:148]:
```suggestion
      points={{0,40},{0,36},{0,30}},
```

File: ThermofluidStream/FlowControl/FourWaySwitch.mo [hunk 14]
```diff
@@ -1,26 +1,47 @@
...
+        rotation=180,
+        origin={-107.5,-27.5})));
+  Modelica.Blocks.Interfaces.RealInput u
+    annotation (Placement(transformation(
+        extent={{-20,-20},{20,20}},
+        rotation=90,
+        origin={0,-100}), iconTransformation(
+        extent={{-20,-20},{20,20}},
+        rotation=90,
+        origin={0,-100})));
```
Comment (tobolar) [inline ThermofluidStream/FlowControl/FourWaySwitch.mo:44]:
```suggestion
        origin={0,-100})));
```

File: ThermofluidStream/FlowControl/Tests/MCV.mo [hunk 15]
```diff
@@ -234,7 +234,7 @@ equation
       points={{10,0},{20,0}},
       color={28,108,200},
       thickness=0.5));
-  connect(pulse1.y, source1.p0_var) annotation (Line(points={{-59,0},{-40,0},{-40,6},{-32,6}}, color={0,0,127}));
+  connect(pulse1.y, source1.p0_var) annotation (Line(points={{-59,6},{-32,6}},                 color={0,0,127}));
```
Comment (tobolar) [inline ThermofluidStream/FlowControl/Tests/MCV.mo:237]:
```suggestion
  connect(pulse1.y, source1.p0_var) annotation (Line(points={{-59,6},{-32,6}}, color={0,0,127}));
```

File: ThermofluidStream/FlowControl/Tests/MCV.mo [hunk 16]
```diff
@@ -247,8 +247,7 @@ equation
       points={{10,-30},{20,-30}},
       color={28,108,200},
       thickness=0.5));
-  connect(trapezoid.y, source2.p0_var) annotation (Line(points={{-59,-30},{-40,-30},{-40,-24},{-32,-24}},
-                                                                                                      color={0,0,127}));
+  connect(trapezoid.y, source2.p0_var) annotation (Line(points={{-59,-24},{-32,-24}},                 color={0,0,127}));
```
Comment (tobolar) [inline ThermofluidStream/FlowControl/Tests/MCV.mo:250]:
```suggestion
  connect(trapezoid.y, source2.p0_var) annotation (Line(points={{-59,-24},{-32,-24}}, color={0,0,127}));
```

File: ThermofluidStream/FlowControl/Tests/MCV.mo [hunk 17]
```diff
@@ -261,7 +260,7 @@ equation
       points={{10,-60},{20,-60}},
       color={28,108,200},
       thickness=0.5));
-  connect(pulse3.y, source3.p0_var) annotation (Line(points={{-59,-60},{-40,-60},{-40,-54},{-32,-54}}, color={0,0,127}));
+  connect(pulse3.y, source3.p0_var) annotation (Line(points={{-59,-54},{-32,-54}},                     color={0,0,127}));
```
Comment (tobolar) [inline ThermofluidStream/FlowControl/Tests/MCV.mo:263]:
```suggestion
  connect(pulse3.y, source3.p0_var) annotation (Line(points={{-59,-54},{-32,-54}}, color={0,0,127}));
```

File: ThermofluidStream/FlowControl/Tests/MCV.mo [hunk 18]
```diff
@@ -274,7 +273,7 @@ equation
       points={{10,-90},{20,-90}},
       color={28,108,200},
       thickness=0.5));
-  connect(trapezoid1.y, source4.p0_var) annotation (Line(points={{-57,-90},{-40,-90},{-40,-84},{-32,-84}}, color={0,0,127}));
+  connect(trapezoid1.y, source4.p0_var) annotation (Line(points={{-57,-84},{-32,-84}},                     color={0,0,127}));
```
Comment (tobolar) [inline ThermofluidStream/FlowControl/Tests/MCV.mo:276]:
```suggestion
  connect(trapezoid1.y, source4.p0_var) annotation (Line(points={{-57,-84},{-32,-84}}, color={0,0,127}));
```

File: ThermofluidStream/Sensors/MultiSensor_Tpm.mo [hunk 19]
```diff
@@ -8,44 +11,51 @@ model MultiSensor_Tpm "Sensor for Temperature, pressure and mass-flow"
...
-        iconTransformation(extent={{80,-20},{120,20}})));
-  Modelica.Blocks.Interfaces.RealOutput m_flow_out(unit="kg/s") = m_flow if outputMassFlowRate "Measured massflow [kg/s]"
-    annotation (Placement(transformation(extent={{80,-80},{120,-40}})));
+  Modelica.Blocks.Interfaces.RealOutput T_out(final quantity="ThermodynamicTemperature", final unit=temperatureUnit) = T if outputTemperature "Temperature output connector"
+    annotation (Placement(transformation(extent={{72,50},{92,70}}), iconTransformation(extent={{72,50},{92,70}})));
+  Modelica.Blocks.Interfaces.RealOutput p_out(final quantity="Pressure", final unit=pressureUnit) = p if outputPressure "Pressure output connector"
+    annotation (Placement(transformation(extent={{72,-10},{92,10}}),
+        iconTransformation(extent={{72,-10},{92,10}})));
+  Modelica.Blocks.Interfaces.RealOutput m_flow_out(unit="kg/s") = m_flow if outputMassFlowRate "Mass flow rate output connector"
+    annotation (Placement(transformation(extent={{72,-70},{92,-50}}), iconTransformation(extent={{72,-70},{92,-50}})));
```
Comment (tobolar) [inline ThermofluidStream/Sensors/MultiSensor_Tpm.mo:58]:
```suggestion
    annotation (Placement(transformation(extent={{72,-70},{92,-50}})));
```

File: ThermofluidStream/Sensors/MultiSensor_Tpm.mo [hunk 20]
```diff
@@ -8,44 +11,51 @@ model MultiSensor_Tpm "Sensor for Temperature, pressure and mass-flow"
...
     annotation (Placement(transformation(extent={{80,-120},{120,-80}})));
-  Modelica.Blocks.Interfaces.RealOutput T_out(final quantity="ThermodynamicTemperature", final unit=temperatureUnit) = T if outputTemperature "Measured Temperature [variable]"
-    annotation (Placement(transformation(extent={{80,40},{120,80}})));
-  Modelica.Blocks.Interfaces.RealOutput p_out(final quantity="Pressure", final unit=pressureUnit) = p if outputPressure "Measured pressure [variable]"
-    annotation (Placement(transformation(extent={{80,-20},{120,20}}),
-        iconTransformation(extent={{80,-20},{120,20}})));
-  Modelica.Blocks.Interfaces.RealOutput m_flow_out(unit="kg/s") = m_flow if outputMassFlowRate "Measured massflow [kg/s]"
-    annotation (Placement(transformation(extent={{80,-80},{120,-40}})));
+  Modelica.Blocks.Interfaces.RealOutput T_out(final quantity="ThermodynamicTemperature", final unit=temperatureUnit) = T if outputTemperature "Temperature output connector"
+    annotation (Placement(transformation(extent={{72,50},{92,70}}), iconTransformation(extent={{72,50},{92,70}})));
```
Comment (tobolar) [inline ThermofluidStream/Sensors/MultiSensor_Tpm.mo:53]:
```suggestion
    annotation (Placement(transformation(extent={{72,50},{92,70}})));
```

File: ThermofluidStream/Sensors/MultiSensor_Tpm.mo [hunk 21]
```diff
@@ -8,44 +11,51 @@ model MultiSensor_Tpm "Sensor for Temperature, pressure and mass-flow"
...
+  Modelica.Blocks.Interfaces.RealOutput p_out(final quantity="Pressure", final unit=pressureUnit) = p if outputPressure "Pressure output connector"
+    annotation (Placement(transformation(extent={{72,-10},{92,10}}),
+        iconTransformation(extent={{72,-10},{92,10}})));
```
Comment (tobolar) [inline ThermofluidStream/Sensors/MultiSensor_Tpm.mo:56]:
```suggestion
    annotation (Placement(transformation(extent={{72,-10},{92,10}})));
```

**Comment by nieweber** [comment]
Regression test passed.

**Pull request merged**